lib/stack_master/sparkle_formation/user_data_file.rb

```diff
@@ -0,0 +1,71 @@
+require 'sparkle_formation'
+require 'erubis'
+
+class SparkleFormation
+  module SparkleAttribute
+    module Aws
+      class SfEruby < Erubis::Eruby
+        include Erubis::ArrayEnhancer
+
+        def add_expr(src, code, indicator)
+          case indicator
+            when '='
+              src << " #{@bufvar} << (" << code << ');'
+            else
+              super
+          end
+        end
+      end
+
+      class TemplateContext < AttributeStruct
+        include SparkleAttribute
+        include SparkleAttribute::Aws
+        include Utils::TypeCheckers
+
+        def initialize(vars)
+          vars.each do |key, value|
+            self.class.send(:define_method, key) do
+              value
+            end
+          end
+          self._camel_keys = true
+        end
+      end
+
+      UserDataFileNotFound = Class.new(StandardError)
+
+      def _user_data_file(file_name, vars = {})
+        file_path = File.join(::SparkleFormation.sparkle_path, 'user_data', file_name)
+        template = File.read(file_path)
+        template_context = TemplateContext.new(vars)
+        compiled_template = SfEruby.new(template).evaluate(template_context)
```

Comment:
With monkey patched `SparkleFormation::SparkleStruct` we could just `.evaluate(self)`.